File: PaperLiteratureManagement/backend/src/main/java/com/paper/entity/Tag.java
package com.paper.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "tags")
public class Tag {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 100)
    private String name;

    @Column(length = 20)
    private String color = "#3b82f6";

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
}

File: PersonalBlogCMS/api/entities/Tag.ts
import { Entity, Column } from 'typeorm';
import { Base } from './Base.js';

@Entity('tags')
export class Tag extends Base {
  @Column({ unique: true, length: 50 })
  name: string;

  @Column({ unique: true, length: 100 })
  slug: string;

  @Column({ length: 20, default: '#10b981' })
  color: string;

  @Column({ name: 'article_count', default: 0 })
  articleCount: number;
}
